packages/flutter/lib/src/widgets/image.dart

```diff
@@ -660,17 +755,26 @@ class _ImageState extends State<Image> {
       _imageStream.removeListener(_handleImageChanged);
 
     if (!widget.gaplessPlayback)
-      setState(() { _imageInfo = null; });
+      setState(() {
+        _chunkEvent = null;
+        _imageInfo = null;
+      });
 
     _imageStream = newStream;
     if (_isListeningToStream)
-      _imageStream.addListener(_handleImageChanged);
+      _imageStream.addListener(
+        _handleImageChanged,
+        chunkListener: widget.imageLoadingBuilder != null ? _handleImageChunkLoaded : null,
```

Comment:
as far as i can tell you don't handle widget.imageLoadingBuilder changing over time